src/mqt/problemsolver/partialcompiler/Partial_QAOA.py

```diff
@@ -0,0 +1,192 @@
+        if num_qubits <= manila_config.n_qubits:
+            self.coupling_map = manila_config.coupling_map
+            self.device_qubits = manila_config.n_qubits
+        elif num_qubits <= montreal_config.n_qubits:
+            self.coupling_map = montreal_config.coupling_map
+            self.device_qubits = montreal_config.n_qubits
+        elif num_qubits <= washington_config.n_qubits:
+            self.coupling_map = washington_config.coupling_map
+            self.device_qubits = washington_config.n_qubits
+
+    def get_uncompiled_circuit(
+        self, include_online_edges: bool = False
+    ) -> tuple[QuantumCircuit, list[QuantumCircuit], list[QuantumCircuit]]:
+        """Return the state preparation circuit and lists of problem and mixer circuits without any compilation.
+         Online edges are optionally included.
+
+        :param include_online_edges: Include online edges in the problem circuits
+        """
+        qc_prep = QuantumCircuit(self.num_qubits)
+        qc_prep.h(range(self.num_qubits))
+        qc_prep.barrier()
+        qcs_problem = []
+        qcs_mix = []
+        self.problem_parameters = []
+        for i in range(self.repetitions):
+            qc_problem = QuantumCircuit(self.num_qubits)
+            p = Parameter(f"a_{i}")
+            self.problem_parameters.append(p)
+            for elem in self.offline_time_edges:
+                qc_problem.rzz(p, elem[0], elem[1])
+            if include_online_edges:
+                for elem in self.online_time_edges:
+                    qc_problem.rzz(p, elem[0], elem[1])
+
+            qcs_problem.append(qc_problem)
+
+            m = Parameter(f"b_{i}")
+            qc_mix = QuantumCircuit(self.num_qubits)
+            qc_mix.barrier()
+            qc_mix.rx(2 * m, range(self.num_qubits))
+            qc_mix.barrier()
+            qcs_mix.append(qc_mix)
+
+        return qc_prep, qcs_problem, qcs_mix
+
+    def determine_mapping(self) -> list[int]:
+        """Determines a mapping layout to the selected quantum devices based on the offline edges."""
+        qc_prep, qcs_problem_uncompiled, qcs_mix_uncompiled = self.get_uncompiled_circuit()
+        assert len(qcs_problem_uncompiled) == len(qcs_mix_uncompiled)
+        qc_composed = qc_prep.copy()
+        for i in range(len(qcs_problem_uncompiled)):
+            qc_composed.compose(qcs_problem_uncompiled[i], inplace=True)
+            qc_composed.compose(qcs_mix_uncompiled[i], inplace=True)
+
+        offline_mapped_qc = transpile(
+            qc_composed, coupling_map=self.coupling_map, basis_gates=IBM_GATES, optimization_level=3
+        )
+
+        layout = offline_mapped_qc._layout.initial_layout
+        mapping = []
+        for elem in layout.get_virtual_bits():
+            if elem.register.name == "ancilla":
+                pass
+            else:
+                mapping.append(layout.get_virtual_bits()[elem])
+        return mapping
```

Comment:
Because that layout includes the ancilla qubits which should not be present when using the layout to compile a not-yet-mapped circuit, since this leads to the error:
`'FullAncillaAllocation: The layout refers to a qubit that does not exist in circuit.'`

Reply:
Technically, you could augment the circuit with those additional qubits beforehand 🙃 
Although, I'd say that this would be equally complex as the current solution. SO maybe just keep it.

Reply:
I do also think it makes sense to keep it as it is. Most likely this part will change anyways in the next weeks.